modules/generator/processor/hostinfo/processor.go

```diff
@@ -0,0 +1,71 @@
+package hostinfo
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/modules/generator/registry"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"go.opentelemetry.io/otel"
+)
+
+const (
+	Name = "host-info"
+
+	hostInfoMetric     = "traces_host_info"
+	hostIdentifierAttr = "grafana.host.id"
+)
+
+var tracer = otel.Tracer("modules/generator/processor/hostinfo")
+
+type Processor struct {
+	Cfg    Config
+	logger log.Logger
+
+	gauge      registry.Gauge
+	registry   registry.Registry
+	metricName string
+}
+
+func (p *Processor) Name() string {
+	return Name
+}
+
+func (p *Processor) PushSpans(ctx context.Context, req *tempopb.PushSpansRequest) {
+	_, span := tracer.Start(ctx, "hostinfo.PushSpans")
+	defer span.End()
+
+	for i := 0; i < len(req.Batches); i++ {
+		resourceSpan := req.Batches[i]
+		attrs := resourceSpan.GetResource().GetAttributes()
+
+	idAttrLoop:
+		for _, idAttr := range p.Cfg.HostIdentifiers {
+			for _, attr := range attrs {
+				if attr.GetKey() == idAttr {
+					labelValues := p.registry.NewLabelValueCombo(
+						[]string{hostIdentifierAttr},
+						[]string{attr.GetValue().GetStringValue()},
```

Comment:
any concern about this not being a string? seems unlikely

Reply:
unlikely but I made sure to check for this